internal/services/synapse/synapse_sql_pool_resource.go

```diff
@@ -168,7 +169,69 @@ func resourceSynapseSqlPool() *pluginsdk.Resource {
 
 			"tags": tags.Schema(),
 		},
+
+		CustomizeDiff: pluginsdk.CustomizeDiffShim(synapseSqlPoolCustomizeDiff),
+	}
+
+	if !features.FourPointOhBeta() {
+		// NOTE: In v3.0 providers this will be an Optional field with a 'Default'
+		// of 'GRS' to match existing v3.0 behavior, the 'ForceNew' logic will be
+		// applied in the CustomizeDiff function...
+		resource.Schema["storage_account_type"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeString,
+			Default:  string(synapse.StorageAccountTypeGRS),
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(synapse.StorageAccountTypeLRS),
+				string(synapse.StorageAccountTypeGRS),
+			}, false),
+		}
+	} else {
+		resource.Schema["storage_account_type"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(synapse.StorageAccountTypeLRS),
+				string(synapse.StorageAccountTypeGRS),
+			}, false),
+		}
+	}
+
+	return resource
+}
+
+func synapseSqlPoolCustomizeDiff(ctx context.Context, d *pluginsdk.ResourceDiff, v interface{}) error {
+	_, value := d.GetChange("geo_backup_policy_enabled")
+	geoBackupEnabled := value.(bool)
+
+	_, value = d.GetChange("storage_account_type")
+	storageAccountType := synapse.StorageAccountType(value.(string))
+
+	if !features.FourPointOhBeta() {
+		var oldValue interface{}
+
+		// Since the pre-existing resources will not have a value for this field
+		// in the state file, we need to create a fake state entry for it here
+		// which contains the default GRS value.
+		if oldValue, _ = d.GetChange("storage_account_type"); len(oldValue.(string)) == 0 {
+			oldValue = string(synapse.StorageAccountTypeGRS)
+		}
+
+		// If the value changes (e.g., GRS -> LRS) destroy and re-create the resource.
+		// This aligns the v3.0 behavior with the v4.0 behavior while not impacting the
+		// existing resources unless they attempt to change the underlying storage
+		// account type for the Sql Pool...
+		if oldValue.(string) != value.(string) {
+			return d.ForceNew("storage_account_type")
+		}
 	}
```

Comment:
As noted above, we can set this on the schema and don't need to do anything with it here.

Reply:
Fixed.